cmd/main.go

```diff
@@ -97,86 +106,116 @@ func main() {
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
-		os.Exit(1)
+		exitCode = 1
+		return
+	}
+
+	baseClient := garmClient.New(&garmClient.GarmScopeParams{
+		BaseURL:  config.Config.Garm.Server,
+		Username: config.Config.Garm.Username,
+		Password: config.Config.Garm.Password,
+		Debug:    false,
+	})
+
+	err = baseClient.Login()
+	if err != nil {
+		setupLog.Error(err, "unable to Login into GARM Client")
+		exitCode = 1
+		return
```

Comment:
shouldn't it be enough to just exit here (`os.exit(1)`) if we are not allowed to login into garm - and with that, kubernetes will re-start the pod again (with the backoff-interval).

the current behavior of the operator is: that each reconciliation-loop is totally independent and if one reconciler crashes, the other controller loops are still continuing.
With your proposed implementation, an error in any of the reconciliation loops will immediately stop the entire program.

@rafalgalaw what was the intention for changing the current behavior?